scripts/api/article.ts

```diff
@@ -430,7 +437,7 @@ interface IArticleApi {
      */
     isPublished(article: IArticle, includeScheduled?: boolean): boolean;
 
-    itemAction(article: IArticle): {[key: string]: boolean};
+    itemAction(article: IArticle): {[key in IAuthoringActionType]: boolean};
```

Comment:
To avoid repeating the type you could use `ReturnType<typeof itemAction>`. If it was repeated more heavily I'd suggest to create another type like `type IItemActionsResult = {[key in IAuthoringActionType]: boolean}`